etp-backend/src/main/clj/solita/common/map.clj

```diff
@@ -27,5 +27,6 @@
 
 (defn dissoc-in [m ks]
   (if (> (count ks) 1)
-    (update-in m (butlast ks) dissoc (last ks))
+    (if (map? (get-in m (butlast ks)))
```

Comment:
Tämä taitaa nyt palauttaa nillin, jos polun päästä ei löydy mappia. Olisiko turvallisempaa palauttaa m sellaisenaan?

Reply:
Meillä on muuten energiatodistus schemassa dissoc-path, joka tekee melkein saman. Voisiko nuo yhdistää tänne map-nimiavaruuden alle? Dissoc-path lienee parempi nimi, niin on selvempää että se toimii vain mappien kanssa.

Reply:
joo pitää palauttaa m, mutta tuon kyllä pitäisikin palauttaa m sellaisenaan. 

Muutin tuota koska huomasin että jos polkua ei ole olemassa niin se luo polun ko. mappiin.

Reply:
voisin ehkä laittaa m:n omalle riville

Reply:
dissoc-in on ehkä enemmän clojuren nimeämiskäytäntöjen mukaista dissoc-path taas ramdaa

Reply:
get-in, assoc-in, update-in kaikki jossa annetaan keypath on postfix -in

Reply:
mutta siirryn tuolla et skeemassakin käyttäjämään tuota yleistä funktiota

Reply:
dissoc-in siis antaa ymmärtää, että tämä voisi toimia millä tahansa assosiatiivisella tietorakenteella esim. vektorilla, kuten get-in, assoc-in, update-in jne.

Siinä mielessä ajattelin dissoc-pathia, että on selvää että tämä ei toimi minkä tahansa assosiatiivisen tietorakenteen kanssa.

Toisaalta tämä on map-nimiavaruudessa ja ehkä tätä voisi myöhemmin laajentaa toimimaan myös vektoreille.